File: backend/src/application/jobs_search/dto/job_details.py
from dataclasses import dataclass
from datetime import datetime
from uuid import UUID

from domain.company.enums import EmployeeCount
from domain.job_posting.enum import EmploymentType, WorkMode, RelevantWorkExperience, MinimumEducationLevel, Gender, \
    MilitaryServiceStatus, JobPostingStatus


@dataclass
class JobDetails:
    id: UUID
    company_title: str
    company_en_title: str
    company_activity: str
    company_employee_count: EmployeeCount
    city: str
    province: str
    job_category: str
    job_title: str
    job_description: str
    company_overview: str
    employment_type: EmploymentType
    work_mode: WorkMode
    salary_range: str
    work_experience: RelevantWorkExperience
    minimum_education: MinimumEducationLevel
    gender: Gender
    military_status: MilitaryServiceStatus
    post_notifications: bool
    status: JobPostingStatus
    created_at: datetime

File: frontend/src/entities/job-details/model/job-details.ts
export interface JobDetails {
    id: string;
    company_title: string;
    company_en_title: string;
    company_activity: string;
    company_employee_count: EmployeeCount;
    city: string;
    province: string;
    job_category: string;
    job_title: string;
    job_description: string;
    company_overview: string;
    employment_type: EmploymentType;
    work_mode: WorkMode;
    salary_range: string;
    work_experience: WorkExperience;
    minimum_education: EducationLevel;
    gender: Gender;
    military_status: MilitaryStatus;
    post_notifications: boolean;
    status: JobStatus;
    created_at: string;
}

export type EmployeeCount =
    | "1_10"
    | "2_10"
    | "11_50"
    | "51_200"
    | "201_500"
    | "500_plus";

export type EmploymentType =
    | "full_time"
    | "part_time"
    | "internship";

export type WorkMode =
    | "onsite"
    | "remote"
    | "hybrid";

export type WorkExperience =
    | "not_important"
    | "less_than_1"
    | "1_3"
    | "3_5"
    | "5_plus";

export type EducationLevel =
    | "not_important"
    | "diploma"
    | "associate"
    | "bachelor"
    | "master"
    | "phd";

export type Gender =
    | "not_important"
    | "male"
    | "female";

export type MilitaryStatus =
    | "not_important"
    | "completed"
    | "exempt";

export type JobStatus =
    | "active"
    | "inactive"
    | "draft";
